dashboard/utils.py

```diff
@@ -38,6 +44,7 @@ class MMTrack:
     financial_aid_min_price = None
     financial_aid_max_price = None
     financial_aid_date_documents_sent = None
+    final_grades = {}
```

Comment:
Based on the code in the `extract_final_grade` method, this is actually a cache. Can you use [Django's cache framework](https://docs.djangoproject.com/en/1.10/topics/cache/) instead of implementing a cache yourself?